doc/user/content/manage/troubleshooting.md

```diff
@@ -10,104 +10,340 @@ aliases:
+
+Materialize dataflows act on collections of data. To provide fast access to the changes to individual records, the records can be stored in an indexed representation called [arrangements](https://materialize.com/docs/get-started/arrangements/#arrangements). Arrangements can be manually created by users on views by creating an index on the view. But they are also used internally in dataflows, for instance, when joining relations.
+
+
+### Translating SQL to dataflows
+
+To make these concepts a bit more tangible, let's look at the example from the [getting started guide](https://materialize.com/docs/get-started/quickstart/).
+
+```sql
+CREATE SOURCE auction_house
+  FROM LOAD GENERATOR AUCTION
+  (TICK INTERVAL '100ms')
+  FOR ALL TABLES
+  WITH (SIZE = '3xsmall');
+
+CREATE MATERIALIZED VIEW num_bids AS
+  SELECT auctions.item, count(bids.id) AS number_of_bids
+  FROM bids
+  JOIN auctions ON bids.auction_id = auctions.id
+  WHERE bids.bid_time < auctions.end_time
+  GROUP BY auctions.item;
+
+CREATE INDEX num_bids_idx ON num_bids (item);
+```
+
+The query of the materialized view joins the relations `bids` and `auctions`, groups by `auctions.item` and determines the number of bids per auction. To understand how this SQL query is translated to a dataflow, we can use `EXPLAIN` to display the plan used to evaluate the join.
+Note that you can also explain the plan of queries and common views with `EXPLAIN` and `EXPLAIN VIEW`, respectively.
+
+```sql
+EXPLAIN MATERIALIZED VIEW num_bids
+```
+```
+                Optimized Plan
+-----------------------------------------------
+ materialize.public.num_bids:                 +
+   Reduce group_by=[#0] aggregates=[count(*)] +
+     Project (#3)                             +
+       Filter (#1 < #4)                       +
+         Join on=(#0 = #2) type=differential  +
+           ArrangeBy keys=[[#0]]              +
+             Project (#2, #4)                 +
+               Get materialize.public.bids    +
+           ArrangeBy keys=[[#0]]              +
+             Project (#0, #2, #3)             +
+               Get materialize.public.auctions+
+
+(1 row)
+```
 
-## Limitations
+The plan describes the specific operators that are used to evaluate the query.
+Some of these operators resemble relational algebra or map reduce style operators (`Filter`, `Join on`, `Project`).
+Others are specific to Materialize (`Get`, `ArrangeBy`).
 
+In general, a high level understanding of what these operators do is sufficient for effective debugging:
+`Filter` filters records, `Join on` joins records from two or more inputs, `Map` applies a function to transform records, etc.
+The `ArrangeBy` operator indexes data and stores it in memory for fast point lookups.
```

Comment:
`ArrangeBy` is a bit tricky, because in many cases it is just signaling that an existing arrangement will be used. So if there is an `ArrangeBy` in the plan, it might or might not build an arrangement as part of this dataflow.

Reply:
So, I'd suggest to just skip trying to explain `ArrangeBy` for now.

Btw. I have some plans to make EXPLAIN print an `ArrangeBy` differently based on whether it's reading an existing index or creating a new arrangement. So later we can clarify this.

Reply:
You are right, talking about the nuances may be too much. I've removed the discussion for now. The PR looks interesting!